eclipse-collections/src/main/java/org/eclipse/collections/impl/map/mutable/ConcurrentHashMap.java

```diff
@@ -965,9 +967,9 @@ else if (o != null)
     }
 
     @Override
-    public Set<K> keySet()
+    public MutableSet<K> keySet()
     {
-        return new KeySet();
+        return SetAdapter.adapt(new KeySet());
```

Comment:
@johnjstuart can't `KeySet` directly implement `MutableSet` since it is implemented in the same file?

Reply:
@johnjstuart you can use `Sets.adapt()` here.